components/brave_wallet/browser/solana_transaction.cc

```diff
@@ -274,16 +275,24 @@ SolanaTransaction::GetSignedTransactionBytes(
   // for this transaction.
   uint8_t num_of_sig = 0;
   for (const auto& signer : signers) {
-    if (*selected_account == signer) {
+    if (base::EqualsCaseInsensitiveASCII(selected_account->address, signer)) {
       if (selected_account_signature) {
         transaction_bytes.insert(transaction_bytes.end(),
                                  selected_account_signature->begin(),
                                  selected_account_signature->end());
       } else {
-        std::vector<uint8_t> signature =
-            keyring_service->SignMessageBySolanaKeyring(signer, message_bytes);
-        transaction_bytes.insert(transaction_bytes.end(), signature.begin(),
-                                 signature.end());
+        for (auto& acc : all_accounts->accounts) {
```

Comment:
This whole else block is not right, we should be calling `SignMessageBySolanaKeyring` with `selected_account->address` directly. Also selected account already being granted permission before the code flow reaches here and there won't be any missing account because signer == selected account. The only difference here would be selected account signature is prefilled or not. (In that case, hardware wallet)

Reply:
fixed in https://github.com/brave/brave-core/pull/18930/commits/b6ea0b849be572662571b3ed3946440507c1124e